From ae09e0aab3feed6ca6fb24da7ea5e19b7c65c4fa Mon Sep 17 00:00:00 2001
From: Andreas Scherer <andreas_tex@freenet.de>
Date: Fri, 26 Jul 2024 09:55:07 +0000
Subject: [PATCH] [PDFTEX] Prepare for compilation with gcc-14.

There's a type mismatch (probably copy-paste-error) for
'pdf_font_has_space_char', defined as '^boolean' in section 821,
in sections 1502 and 1518.

See https://tug.org/pipermail/tex-live/2024-July/050773.html for the
initial bug report.

git-svn-id: svn://tug.org/texlive/trunk/Build/source@71901 c570f23f-e606-0410-a88d-b1316a301751

diff --git a/texk/web2c/pdftexdir/pdftex.ch b/texk/web2c/pdftexdir/pdftex.ch
index 0607f7f2f1..a30a74c337 100644
--- texk/web2c/pdftexdir/pdftex.ch
+++ texk/web2c/pdftexdir/pdftex.ch
@@ -179,7 +179,7 @@ pdf_font_type:=xmalloc_array(eight_bits, font_max);
 pdf_font_attr:=xmalloc_array(str_number, font_max);
 pdf_font_blink:=xmalloc_array(internal_font_number, font_max);
 pdf_font_elink:=xmalloc_array(internal_font_number, font_max);
-pdf_font_has_space_char:=xmalloc_array(internal_font_number, font_max);
+pdf_font_has_space_char:=xmalloc_array(boolean, font_max);
 pdf_font_stretch:=xmalloc_array(integer, font_max);
 pdf_font_shrink:=xmalloc_array(integer, font_max);
 pdf_font_step:=xmalloc_array(integer, font_max);
@@ -298,7 +298,7 @@ pdf_font_type:=xmalloc_array(eight_bits,font_max);
 pdf_font_attr:=xmalloc_array(str_number,font_max);
 pdf_font_blink:=xmalloc_array(internal_font_number,font_max);
 pdf_font_elink:=xmalloc_array(internal_font_number,font_max);
-pdf_font_has_space_char:=xmalloc_array(internal_font_number,font_max);
+pdf_font_has_space_char:=xmalloc_array(boolean,font_max);
 pdf_font_stretch:=xmalloc_array(integer,font_max);
 pdf_font_shrink:=xmalloc_array(integer,font_max);
 pdf_font_step:=xmalloc_array(integer,font_max);
